extension/indexer/standard_indexer.go

```diff
@@ -0,0 +1,210 @@
+	v, err := s.db.Get(txID[:])
+	if errors.Is(err, database.ErrNotFound) {
+		return false, 0, false, fees.Dimensions{}, 0, nil
+	}
+	if err != nil {
+		return false, 0, false, fees.Dimensions{}, 0, err
+	}
+	timestamp := int64(binary.BigEndian.Uint64(v))
+	success := true
+	if v[consts.Uint64Len] == failureByte {
+		success = false
+	}
+	d, err := fees.UnpackDimensions(v[consts.Uint64Len+1 : consts.Uint64Len+1+fees.DimensionsLen])
+	if err != nil {
+		return false, 0, false, fees.Dimensions{}, 0, err
+	}
+	fee := binary.BigEndian.Uint64(v[consts.Uint64Len+1+fees.DimensionsLen:])
+	return true, timestamp, success, d, fee, nil
+}
+
+func (s *StandardDBIndexer) Accepted(_ context.Context, blk *chain.StatelessBlock) error {
+	batch := s.db.NewBatch()
+	defer batch.Reset()
+
+	// Store block bytes
+	if err := s.storeBlock(batch, blk); err != nil {
+		return err
+	}
+	// Map block ID to block height
+	if err := s.mapBlockIDToHeight(batch, blk.ID(), blk.Hght); err != nil {
+		return err
+	}
+	// Store TX
+	timestamp := blk.GetTimestamp()
+	results := blk.Results()
+	for j, tx := range blk.Txs {
+		result := results[j]
+		if err := s.storeTransaction(
+			batch,
+			tx.ID(),
+			timestamp,
+			result.Success,
+			result.Units,
+			result.Fee,
+		); err != nil {
+			return err
+		}
+	}
+
+	return batch.Write()
+}
+
+func (s *StandardDBIndexer) AcceptedStateful(_ context.Context, blk *chain.StatefulBlock) error {
+	batch := s.db.NewBatch()
+	defer batch.Reset()
+
+	// Store block itself
+	if err := s.storeStatefulBlock(batch, blk); err != nil {
+		return err
+	}
+	blkID, err := blk.ID()
+	if err != nil {
+		return err
+	}
+	// Map block ID to block height
+	if err = s.mapBlockIDToHeight(batch, blkID, blk.Hght); err != nil {
+		return err
+	}
+	// Store TX
+	timestamp := blk.Tmstmp
+	for _, tx := range blk.Txs {
+		if err := s.storeTransactionStateful(batch, tx.ID(), timestamp); err != nil {
+			return err
+		}
+	}
+
+	return batch.Write()
+}
+
+func (s *StandardDBIndexer) GetBlockByHeight(height uint64) ([]byte, error) {
+	k := make([]byte, consts.Uint64Len)
+	binary.BigEndian.PutUint64(k, height)
+	v, err := s.db.Get(k)
+	if errors.Is(err, database.ErrNotFound) {
+		return []byte{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return v, nil
+}
+
+func (s *StandardDBIndexer) GetBlock(blockID ids.ID) ([]byte, error) {
+	// Get height
+	v, err := s.db.Get(blockID[:])
+	if errors.Is(err, database.ErrNotFound) {
+		return []byte{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// Get block
+
+	v, err = s.db.Get(v)
+	if errors.Is(err, database.ErrNotFound) {
+		return []byte{}, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	return v, nil
+}
+
+func (s *StandardDBIndexer) BlockAlreadyIndexed(height uint64) bool {
+	k := make([]byte, consts.Uint64Len)
+	binary.BigEndian.PutUint64(k, height)
+	exists, _ := s.db.Has(k)
+	return exists
+}
+
+/*
+Helper storage functions
+*/
+
+func (*StandardDBIndexer) storeTransactionStateful(
```

Comment:
Can we be consistent with the naming of these functions instead of using both `store` and `map` as the prefixes, could we just use `put...` ?